driver-core/src/main/com/mongodb/event/ConnectionPoolEventMulticaster.java

```diff
@@ -16,92 +16,151 @@
 
 package com.mongodb.event;
 
-import com.mongodb.annotations.Beta;
+import com.mongodb.annotations.Immutable;
+import com.mongodb.diagnostics.logging.Logger;
+import com.mongodb.diagnostics.logging.Loggers;
 
-import java.util.Set;
-import java.util.concurrent.ConcurrentHashMap;
+import java.util.ArrayList;
+import java.util.List;
 
-import static java.util.Collections.newSetFromMap;
+import static com.mongodb.assertions.Assertions.isTrue;
+import static com.mongodb.assertions.Assertions.notNull;
+import static java.lang.String.format;
+import static java.util.Collections.unmodifiableList;
 
 /**
- * A multicaster for connection pool events.
+ * A multicaster for connection pool events. Any exception thrown by one of the listeners will be caught and not re-thrown, but may be
+ * logged.
+ *
+ * @since 3.5
  */
-@Beta
+@Immutable
```

Comment:
Should we instead deprecate this class, since ServerSettings and MongoClientOptions.Builder take a List<ConnectionPoolListener> and so there's no need for applications to use a multicaster?